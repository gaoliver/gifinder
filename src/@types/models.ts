export type GIFProps = {
  id: string;
  url: string;
  slug: string;
  embed_url: string;
  source: string;
  title: string;
  rating: string;
  images: {
    original: {
      height: string;
      width: string;
      size: string;
      url: string;
      mp4_size: string;
      mp4: string;
      webp_size: string;
      webp: string;
      frames: string;
      hash: string;
    };
    downsized: {
      height: string;
      width: string;
      size: string;
      url: string;
    };
    downsized_still: {
      height: string;
      width: string;
      size: string;
      url: string;
    };
  };
};

export type SearchResultsApi = {
  data: GIFProps[];
  pagination: {
    total_count: number;
    count: number;
    offset: number;
  };
  meta: {
    status: number;
    msg: string;
    response_id: string;
  };
};
