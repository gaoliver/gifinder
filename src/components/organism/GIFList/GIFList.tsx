import React from 'react';
import { AspectRatio, Box, FlatList, Pressable } from 'native-base';
import { spaces } from '@/constants/spaces';
import { GIFProps } from '@/@types/models';
import FastImage from 'react-native-fast-image';

interface GIFListProps {
  list: GIFProps[];
  onPressItem: (item: GIFProps) => void;
  onEndReached?: () => void;
  onEndReachedThreshold?: number;
  ListFooterComponent?:
    | React.ComponentType<any>
    | React.ReactElement<any, string | React.JSXElementConstructor<any>>;
}

export const GIFList: React.FC<GIFListProps> = ({
  list,
  onPressItem,
  ...props
}) => {
  return (
    <FlatList
      {...props}
      data={list}
      pt={1}
      px={spaces.screenWidth}
      numColumns={3}
      columnWrapperStyle={{ gap: 4 }}
      renderItem={({ item }) => (
        <Box my={1} key={item.id} flex={1 / 3}>
          <Pressable onPress={() => onPressItem(item)} testID={item.id}>
            <AspectRatio ratio={1 / 1} w={'100%'}>
              <FastImage
                source={{
                  uri: item.images.downsized.url,
                }}
              />
            </AspectRatio>
          </Pressable>
        </Box>
      )}
    />
  );
};
