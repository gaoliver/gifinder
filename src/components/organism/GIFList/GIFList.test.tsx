import React from 'react';
import renderer from 'react-test-renderer';
import { GIFProps } from '@/@types/models';
import { NativeBaseProvider } from 'native-base';
import { GIFList } from './GIFList';
test('GIFList renders correctly.', () => {
  const gifData: GIFProps[] = [
    {
      id: 'gif123',
      slug: 'test-gif',
      embed_url: 'https://example.com/embed',
      source: 'https://example.com/source',
      images: {
        original: {
          url: 'https://example.com/image.gif',
          height: '',
          width: '',
          size: '',
          mp4_size: '',
          mp4: '',
          webp_size: '',
          webp: '',
          frames: '',
          hash: '',
        },
        downsized: {
          height: '',
          width: '',
          size: '',
          url: '',
        },
      },
      title: 'Test GIF',
      url: 'https://example.com',
      rating: 'PG',
    },
  ];

  const tree = renderer
    .create(
      <NativeBaseProvider>
        <GIFList list={gifData} onPressItem={() => {}} />
      </NativeBaseProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
