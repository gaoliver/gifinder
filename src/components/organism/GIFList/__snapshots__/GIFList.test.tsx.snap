// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIFList renders correctly. 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
